src/lib/Liblog/pbs_log.c

```diff
@@ -978,84 +1026,66 @@ log_record(int eventtype, int objclass, int sev, const char *objname, const char
 	if ((text == NULL) || (objname == NULL))
 		goto sigunblock;
 
-	/* if gettimeofday() fails, log messages will be printed at the epoch */
-	if (gettimeofday(&tp, NULL) != -1) {
-		now = tp.tv_sec;
+	/* lock the file mutex */
+	if (log_mutex_lock() == 0) {
+		get_timestamp(&mst);
+		
+		/* Do we need to switch the log? */
+		if (log_auto_switch && (mst.ptm.tm_yday != log_open_day)) {
+			log_close(1);
+			log_open(NULL, log_directory);
+			if (log_opened < 1) {
+				log_mutex_unlock();
+				log_console_error("PBS cannot open its log");
+				goto sigunblock;
+			}
+		}
 
-		if (pbs_log_highres_timestamp)
-			snprintf(microsec_buf, sizeof(microsec_buf), ".%06ld", (long)tp.tv_usec);
+		/* call the inner routine which does not lock */
+		log_record_inner(eventtype, objclass, sev, objname, text, &mst);
+		log_mutex_unlock();
 	}
 
-#ifdef WIN32
-	ptm = localtime(&now);
-#else
-	ptm = localtime_r(&now, &ltm);
+sigunblock:
+#ifndef WIN32
+	sigprocmask(SIG_SETMASK, &old_mask, NULL);
 #endif
```

Comment:
I think you will need to add below code here for some platform else it will give compile error:
```c
#else
     return;
#endif
```

Reply:
its a void function, so no need to add return